libraries/abstractions/ota_pal_psa/aws_ota_pal.c

```diff
@@ -0,0 +1,512 @@
+/*
+ * FreeRTOS OTA V1.2.1
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ * Copyright (c) 2021 Arm Limited. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/**
+ * @file aws_ota_pal.c
+ * @brief Platform Abstraction layer for AWS OTA
+ *
+ */
+
+/* C Runtime includes. */
+#include <stdlib.h>
+#include <string.h>
+
+/* Amazon FreeRTOS include. */
+#include "FreeRTOS.h"
+#include "FreeRTOSConfig.h"
+#include "aws_iot_ota_pal.h"
```

Comment:
Just heads up on OTA pal we have minor changes in the interface in the latest release of OTA library and pal  - https://freertos.org/Documentation/api-ref/ota-for-aws-iot-embedded-sdk/docs/doxygen/output/html/ota_pal_interface.html#ota_pal_interface_overview

Reply:
Okay. I will check the update on the latest release.

Reply:
I checked the latest master branch(5251eeaf01e09d24a666bec409bb45d8165a869e), the OTA pal APIs are still defined in aws_iot_ota_pal.h. I checked git log of the ./libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c, the lastest commit was on 4 Nov 2020. So I want to double confirm with @pvyawaha , has the latest release of the OTA library you mentioned been merged into the Amazon FreeRTOS repo?